src/protocols/ethernet.rs

```diff
@@ -1,15 +1,29 @@
-use super::super::utils;
+use std::mem::size_of;
+use crate::utils::bytes_of_mut;
 
 use super::arp;
 use super::ipv4;
 
+#[derive(Debug, Default, Clone, Copy)]
 #[repr(C, packed)]
 pub struct EthernetHeader {
     pub dhost: [u8; 6],
     pub shost: [u8; 6],
     pub ether_type: u16,
 }
 
+impl EthernetHeader {
+	pub fn from_bytes(bytes: &[u8]) -> Option<EthernetHeader> {
+	  if bytes.len() == size_of::<EthernetHeader>() {
```

Comment:
```suggestion
	  if bytes.len() >= size_of::<EthernetHeader>() {
```
Same as ArpHeader

Reply:
Doing that would make the code fail, the byte slice given is only used here to extract the header, it's the job of the caller [to split the slice by the size of the Header](https://doc.rust-lang.org/std/primitive.slice.html#method.split_at).

Reply:
You must split the slice of bytes [in your decode function](https://github.com/Lsh0x/EagleEyes/pull/18/files#diff-6d1443e173a8cddcab0d0e73b3106b98364ce747d8a290478e57edc0f3a080b6R31-R33) to be able to extract the header and the data separately.

```rust
let (left, right) = slice.split_at(size_of::<ArpHeader>());
let header = AprHeader::from_bytes(left);
let tail = right; // those bytes contain the data and maybe more headers?
```

Reply:
If i understand correctly, [this](https://github.com/Lsh0x/EagleEyes/pull/18/files#diff-6d1443e173a8cddcab0d0e73b3106b98364ce747d8a290478e57edc0f3a080b6R32) will not work and must be rewrite to just send the slice of bytes?

Reply:
I fixed that in the latest commit, I parse the first header from the data source, I let you use the tail as you like.